packages/cli/test/integration.js

```diff
@@ -3675,3 +3675,140 @@ test('[vc link] should support the `--project` flag', async t => {
     formatOutput(output)
   );
 });
+
+test('vercel.json projectSettings should apply to prompts', async t => {
+  const directory = path.join(__dirname, 'fixtures/unit/vercel-json-overrides');
+  const vercelJsonPath = path.join(directory, 'vercel.json');
+  const vercelJsonOriginal = await readFile(vercelJsonPath, 'utf8');
+
+  function assertDeployment(deployment, statusCode) {
+    t.is(
+      deployment.exitCode,
+      statusCode,
+      formatOutput({
+        stderr: deployment.stderr,
+        stdout: deployment.stdout,
+      })
+    );
+  }
+
+  async function deploy() {
+    const deployment = await execa(binaryPath, [
+      directory,
+      ...defaultArgs,
+      '--public',
+    ]);
+
+    return deployment;
+  }
+
+  // Deploy without projectSettings
+  await writeFile(vercelJsonPath, JSON.stringify({}));
+  assertDeployment(await deploy(), 0);
```

Comment:
You also need to visit the deployment to see if the output directory was assigned properly. It should fetch the root index file (you could use index.txt instead of index.html to make this a simple content comparison)